Application/Transfer/Import/Standard/Service.php

```diff
@@ -0,0 +1,1561 @@
+use SPHERE\Application\People\Person\Person;
+use SPHERE\Application\People\Person\Service\Entity\TblPerson;
+use SPHERE\Application\People\Person\Service\Entity\TblSalutation;
+use SPHERE\Application\People\Relationship\Relationship;
+use SPHERE\Application\People\Relationship\Service\Entity\TblType as TblTypeRelationship;
+use SPHERE\Common\Frontend\Form\IFormInterface;
+use SPHERE\Common\Frontend\Form\Structure\FormColumn;
+use SPHERE\Common\Frontend\Form\Structure\FormGroup;
+use SPHERE\Common\Frontend\Form\Structure\FormRow;
+use SPHERE\Common\Frontend\Layout\Repository\Accordion;
+use SPHERE\Common\Frontend\Layout\Repository\Listing;
+use SPHERE\Common\Frontend\Layout\Repository\Panel;
+use SPHERE\Common\Frontend\Layout\Structure\Layout;
+use SPHERE\Common\Frontend\Layout\Structure\LayoutColumn;
+use SPHERE\Common\Frontend\Layout\Structure\LayoutGroup;
+use SPHERE\Common\Frontend\Layout\Structure\LayoutRow;
+use SPHERE\Common\Frontend\Message\Repository\Danger;
+use SPHERE\Common\Frontend\Message\Repository\Success;
+use SPHERE\Common\Frontend\Message\Repository\Warning;
+use SPHERE\Common\Frontend\Text\Repository\Danger as DangerText;
+use SPHERE\Common\Frontend\Text\Repository\Muted;
+use SPHERE\Common\Frontend\Text\Repository\Small;
+use SPHERE\Common\Frontend\Text\Repository\Success as SuccessText;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Class Service
+ *
+ * @package SPHERE\Application\Transfer\Import\Standard
+ */
+class Service
+{
+
+    /**
+     * @param IFormInterface|null $Form
+     * @param UploadedFile|null   $File
+     * @param array|null          $Data
+     *
+     * @return string
+     */
+    public function createStudentsFromFile(IFormInterface $Form = null, UploadedFile $File = null, $Data = null)
+    {
+
+        /**
+         * Skip to Frontend
+         */
+        if (null === $File) {
+            return $Form;
+        }
+
+        if ($File->getError()) {
+            $Form->setError('File', 'Fehler');
+            $Form->appendGridGroup(new FormGroup(new FormRow(new FormColumn(new Danger('File nicht gefunden')))));
+            return $Form;
+        }
+        if(isset($Data['Year']) && !empty($Data['Year'])){
+            $YearString = substr($Data['Year'], 0, 4);
+        } else {
+            // fallback
+            $YearString = (new DateTime())->format('Y');
+        }
+
+        /**
+         * Prepare
+         */
+        $File = $File->move($File->getPath(),
+            $File->getFilename().'.'.$File->getClientOriginalExtension());
+
+        /**
+         * Read
+         */
+        //$File->getMimeType()
+        /** @var PhpExcel $Document */
+        $Document = Document::getDocument($File->getPathname());
+        if (!$Document instanceof PhpExcel) {
+            $Form->setError('File', 'Fehler');
+            return $Form;
+        }
+
+        $X = $Document->getSheetColumnCount();
+        $Y = $Document->getSheetRowCount();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $Y. Configured minimum length is 2.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)